### PR TITLE
Spell resistance rework: part 1

### PR DESCRIPTION
**Pull request by rilian-la-te**
This is only immunity replacement part from draft spell lists PR, just for sake to make PRs smaller.

### CODE REVIEW

**Comment by dydzio0614** [comment]
i dont see whats wrong with water elementals being by default immune to water spells from mods TBH

**Comment by rilian-la-te** [comment]
It blocks magic arrow, which is not OH3 behaviour cleanly. So, for part 1 water immunity will be dropped from water elementals.

**Comment by dydzio0614** [comment]
bless, clone, weakness... all work in base H3?

**Comment by rilian-la-te** [comment]
Yes, only Ice Bolt and Frost Ring does not work. Aside normal elemental immunities (NON_LIVING).

**Comment by rilian-la-te** [comment]
Updated to latest develop with battle changes.

**Comment by rilian-la-te** [comment]
@IvanSavenko can you review this part? Psychic and Magic elementals will be in a next part)

File: lib/CBonusTypeHandler.cpp [hunk 1]
```diff
@@ -99,60 +99,66 @@ std::string CBonusTypeHandler::bonusToGraphics(const std::shared_ptr<Bonus> & bo
 		fileName = sp->getIconImmune();
 		break;
 	}
-	case BonusType::FIRE_IMMUNITY:
+	case BonusType::SPELL_DAMAGE_REDUCTION: //Spell damage reduction for all schools
```
Comment (IvanSavenko) [inline lib/CBonusTypeHandler.cpp:102]: And this is why we need way to configure bonus icons at least based on subtype.
Hardcoding everything is not the best idea.

File: client/battle/BattleStacksController.cpp [hunk 2]
```diff
@@ -271,7 +271,16 @@ std::shared_ptr<IImage> BattleStacksController::getStackAmountBox(const CStack *
...
-		effectsPositivness += CGI->spellh->objects.at(spellID)->positiveness;
+	{
+		auto positiveness = CGI->spells()->getByIndex(spellID)->getPositiveness();
+		if(!boost::logic::indeterminate(positiveness))
+		{
+			if(positiveness)
+				effectsPositivness++;
+			else
+				effectsPositivness--;
+		}
```
Comment (IvanSavenko) [inline client/battle/BattleStacksController.cpp:282]: Might be a good idea to replace this with explicit enum instead of tribool

**Review by IvanSavenko — APPROVED** [review]

File: client/battle/BattleStacksController.cpp (hunk 2 above)
Comment (rilian-la-te) [reply client/battle/BattleStacksController.cpp:282]: There is already explicit enum, but used internally. Externally tribool was used. Maybe will do in next PR.

File: lib/CBonusTypeHandler.cpp (hunk 1 above)
Comment (rilian-la-te) [reply lib/CBonusTypeHandler.cpp:102]: Agree. Selector-based config will be after part 2 (psychic and magic elementals).

**Comment by rilian-la-te** [comment]
Updated to latest changes. Again)

**Review by Nordsoft91 — APPROVED** [review]

**Pull request merged**